building for swift 2.3 and Xcode 8

theMeatloaf opened the pull request:
This should allow you to build on Xcode 8 with swift 2.3 to resolve #61.  Swift 3 will need to come next

theMeatloaf committed:
building for swift 2.3 and Xcode 8

theMeatloaf committed:
fixing memory leak on buttons

theMeatloaf committed:
removing memory leaks and cleaning up observers after they are done

ApolloZhu committed:
Swift 3 + Xcode 8.2

ApolloZhu committed:
Remove all observed reference cycles/memory leaking

Finally able to solve? this: LiulietLee/Pick-Color#2
MKLayer’s superLayer observers
MKRefreshControl’s parentScrollView observers
MKSwitch’s parent MKSwitch

It’s sad that when containing MKLayer and MKRefreshControl, current
implementation requires calling `recycle` in `deinit`…

ApolloZhu committed:
Clean Up

serg94 commented:
how to specify this commit in pods if not merged into master?

ApolloZhu commented:
@serg94 
`pod 'MaterialKit', :git => 'https://github.com/theMeatloaf/MaterialKit.git'`
or literally this last commit, add following to the rear
`, :commit => 'a2b6f0422976a2ecf7762e6e08b822a7ca694d6d'`

See: https://guides.cocoapods.org/using/the-podfile.html

theMeatloaf committed:
merging in changes from ApolloZhu

theMeatloaf committed:
I think I may have done it for real this time

ApolloZhu committed:
Grand Swiftify

- Remove self, type delcaration, use lazy init, parameter same as
default
- Merge if-let, use optional changing for short ones, use implicit
unwrapped for guaranteed,
- Replace file private with private, provide public init for an open
class
- Update README

ApolloZhu committed:
Support Swift 4

ApolloZhu committed:
Indicate Swift 4 Support in README

theMeatloaf committed:
merging in new swift changes
